relay-filter/src/config.rs

```diff
@@ -248,13 +251,117 @@ impl GenericFilterConfig {
     }
 }
 
+impl PartialEq for GenericFilterConfig {
+    fn eq(&self, other: &Self) -> bool {
+        self.id == other.id
+    }
+}
+
 /// Configuration for generic filters.
-#[derive(Clone, Debug, Default, Serialize, Deserialize)]
-pub(crate) struct GenericFiltersConfig {
+///
+/// # Deserialization
+///
+/// `filters` is expected to be a [`Vec<GenericfilterConfig>`].
+/// Only the first occurrence of a filter is kept, and duplicates are removed.
+/// Two filters are considered duplicates if they have the same ID,
+/// independently of the condition.
+///
+/// The list of filters is deserialized into an [`IndexMap`], where the key is
+/// the filter's id and the value is the filter itself. The map is converted
+/// back to a list when serializing it, without the filters that were discarded
+/// as duplicates. See examples below.
+///
+/// # Examples
+///
+/// Deserialization:
+///
+/// ```
+/// # use relay_filter::GenericFiltersConfig;
```

Comment:
I decided to add the (de)serialization tests as doctests as the description above may not be 100% clear and I was going to add tests anyway.